src/main/kotlin/ecommerce/dto/products/ProductDTO.kt

```diff
@@ -0,0 +1,16 @@
+package ecommerce.dto.products
+
+import jakarta.validation.constraints.Pattern
+import jakarta.validation.constraints.Size
+import org.hibernate.validator.constraints.Length
+
+data class ProductDTO(
+    @field:Length(min = 3, max = 15, message = "name should be between 3 and 15")
+    @field:Pattern(
+        regexp = "^[a-zA-Z1-9()\\[\\]+\\-&/_]+$",
+        message = "Product name contains invalid characters",
+    )
+    val name: String,
+    @field:Size(min = 1, message = "At least one option required")
+    val optionsList: MutableList<OptionDTO>,
```

Comment:
> Validation and Constraints:<br/>
> * Are there any improvements I can make in the validation strategy (e.g., bean validation, custom constraints) for nested DTOs?
> * How should I validate nested object for DTO, for now I moved the , logic to entity

You can validate nested DTOs by using `@Valid` along with `@field`.

For example:

```suggestion
    @field:Valid
    val optionsList: MutableList<OptionDTO>,
```